src/wallet/wallet.cpp

```diff
@@ -3258,3 +3258,34 @@ ScriptPubKeyMan* CWallet::AddWalletDescriptor(WalletDescriptor& desc, const Flat
 
     return ret;
 }
+
+bool CWallet::GetExtendedKey(const std::string path, CExtKey &ext_key, KeyOriginInfo &info) {
+    if (IsCrypted()) {
+        if (IsLocked()) {
+            return false;
+        }
+        // TODO: decrypt master key if unlocked
+        return false;
+    }
+
+    // TODO: avoid needing a pre-existing ScriptPubKeyMan for obtaining the HD seed
+    auto spk_man = GetScriptPubKeyMan(OutputType::BECH32, true);
+    if (!spk_man) return false;
+    auto desc_spk_man = dynamic_cast<DescriptorScriptPubKeyMan*>(spk_man);
+    CKey seed = desc_spk_man->GetKeys().begin()->second;
+    CExtKey master_key;
+    master_key.SetSeed(seed.begin(), seed.size());
```

Comment:
In addition to being an ugly hack, this is also incorrect and does not actually produce the correct master key.